### PR TITLE
Add `PostTimeout` function

### PR DESCRIPTION
**Pull request by rafrombrc**
This patch makes 4 changes:
- Adds `PostTimeout` function that lets you tell go-notify to time out after a specific time duration for output channels that are blocking. 
- Adds `StopAll` function that lets you unregister all channels listening for a given event.
- Moves the `E_NOT_FOUND` error string to a constant so the output string is only specified in a single location.
- Adds an `outChans` variable that is used to store the results of `events[event]`, so the map lookup doesn't have to happen multiple times unnecessarily.

The `PostTimeout` and `StopAll` functions are both particularly useful from within test code, when you might be exercising code that sets up event listeners that are otherwise beyond control of the tests themselves.

### CODE REVIEW

**Comment by mreiferson** [comment]
thanks @rafrombrc I completely missed this pull request, sorry for the delay...

File: notify.go [hunk 1]
```diff
@@ -43,6 +46,10 @@ var events = make(map[string][]chan interface{})
 // mutex for touching the event map
 var rwMutex sync.RWMutex
 
+// reusable var for storing channel slices to avoid need for multiple map
+// lookups
+var outChans []chan interface{}
```
Comment (mreiferson) [inline notify.go]: i'm not sure I understand the value of making this variable global when it's only used in a local context looking up in the map?

File: notify.go [hunk 2]
```diff
@@ -72,17 +80,55 @@ func Stop(event string, outputChan chan interface{}) error {
...
+	for _, outputChan := range outChans {
 		outputChan <- data
 	}
 
 	return nil
 }
+
+// Post a notification to the specified event using the provided timeout for
+// any output channels that are blocking
+func PostTimeout(event string, data interface{}, timeout *time.Duration) error {
```
Comment (mreiferson) [inline notify.go]: `time.Duration` is in `int64` so I don't think `timeout` should be a pointer

**Comment by mreiferson** [comment]
overall changes look good other than those few nitpicks

**Comment by rafrombrc** [comment]
Yup, good spots. Made the requested changes.

**Comment by mreiferson** [comment]
cool, thanks.  do you mind squashing this down to one commit?

**Comment by mreiferson** [comment]
errr, maybe 3:
1. for the error change
2. for StopAll
3. for PostTimeout

?

**Comment by rafrombrc** [comment]
Commits squashed. First time I've used "git rebase -i"... surprisingly easy!

**Pull request merged**